put loading state and network state to viewmodel as observable fields…

OyaCanli opened the pull request:
… and pass the viewmodel to the binding implementation instead of seperate booleans

rajtheinnovator commented:
@OyaCanli  thanks a lot for fixing this so fast. I wanted to get the whole file before going through the code and will try to do that next day, because it's getting a bit late on my side. Thanks again.

OyaCanli commented:
Just fixed the problem. Using observable fields was not big deal at all and it worked as I expected and fixed the problem with rotation. However I had to change a few other things as well. Previously, fetching was beginning inside the constructor of repository, which is initiated from the constructor of the viewmodel. However, after my recent changes it led to a crash. Because once ViewModel is initialized, repository was initialized, which was followed by connection check and data fetching. Connection check was returning the results right away to ui and binding implementation. But viewmodel was not yet set to the binding at that point, which led to a NPE. So I moved the initialization of fetching away from the constructor of the repository. I initialized it after setting the viewmodel to the binding implementation. And for avoiding double fetching of data, I added a conditional and return inside the method. You'll see when you have time to review the code. Tell me if you have a better idea.

Edit: Sure, whatever you wish.. I was not even expecting you to be awake :)
